From be5a3b1de4611167f76af5f7c95f371c4081151d Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Tue, 4 Aug 2015 20:29:57 -0700
Subject: [PATCH 04/17] input: silead: Added acpi support for power gpio

Added support to read power gpio value from acpi
BIOS.

Change-Id: Ia9fe8eb9d9e78276dce178ab87cbaf680bba18c2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-20919
Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Reviewed-on: https://android.intel.com:443/472777
---
 drivers/input/touchscreen/silead.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/input/touchscreen/silead.c b/drivers/input/touchscreen/silead.c
index 4e0ca7c..e2f61b1 100644
--- a/drivers/input/touchscreen/silead.c
+++ b/drivers/input/touchscreen/silead.c
@@ -468,6 +468,14 @@ static int silead_ts_probe(struct i2c_client *client,
 		data->gpio_power = gpio_to_desc(ret);
 		if (!data->gpio_power)
 			return -ENODEV;
+	} else {
+		data->gpio_power = devm_gpiod_get_index(dev,
+							SILEAD_PWR_GPIO_NAME,
+							1);
+		if (IS_ERR(data->gpio_power)) {
+			dev_err(dev, ">>>>> POWER GPIO request failed\n");
+			return -ENODEV;
+		}
 	}
 
 	ret = gpiod_direction_output(data->gpio_power, 0);
-- 
1.9.1

